asllib/doc/ASLSemanticsReference.tex

```diff
@@ -19,160 +283,1634 @@
+to an \emph{ordering label} $l$, which specifies whether it represents a read or write effect to a variable named $\id$.
+
+An edges represents an ordering constraint between two effects, which can be one of the following:
+\begin{description}
+\item[$\asldata$] Represents a \emph{data dependency}.
+That is, when a write effect reads/write over the data written by another write effect.
+\item[$\aslctrl$] Represents a \emph{control dependency}.
+That is, when a read effect to a variable determines the flow of control (e.g., which condition of a branch is taken),
+which then leads to another read/write effect.
+\item[$\aslpo$] Represents a \emph{program order}.
+That is, when two effects are defined to be ordered according to the sequential semantics of the program.
+\end{description}
+
+An execution graph is \emph{well-formed} if all nodes have unique instance counters, edges connect graph nodes,
+and the output nodes are contained in the set of nodes:
+\[
+  \begin{array}{rcl}
+  \forall n, n' \in N^\xgraph &.& n=(u,l,\id) \wedge n=(u',l',\id') \;\Rightarrow\; u \neq u' \\
+  \forall e \in E^\xgraph &.& e=(n, n', l) \;\Rightarrow\; n, n' \in N^\xgraph \\
+  & & O^\xgraph \subseteq N^\xgraph \enspace.
+  \end{array}
+\]
+
+We denote the empty execution graph $\emptygraph \triangleq (\emptyset, \emptyset, \emptyset)$.
+%
+We define the following functions, which return an execution graph that represents a single read/write effect to a variable \texttt{x}:
+\[
+  \begin{array}{rclll}
+    \WriteEffect(\texttt{x}) &\triangleq& (\{n\}, \emptyset, \{n\}) & \text{where } n=(u, \Write, \texttt{x}), & u\in\N \text{ is fresh}\\
+    \ReadEffect(\texttt{x})  &\triangleq& (\{n\}, \emptyset, \{n\}) & \text{where } n=(u, \Read, \texttt{x}),   & u\in\N \text{ is fresh}\\
+  \end{array}
+\]
+
+We also define two ways to combine execution graphs --- \emph{unordered combination} and \emph{ordered combination}.
+
+Given two execution graphs $S_1 = (N_1, E_1, O_1)$ and $S_1 = (N_2, E_2, O_2)$ their unordered combination,
+denoted $S_1 \parallel S_2$ is defined as follows:
+\[
+  S_1 \parallel S_2\triangleq (N_1 \cup N_2, E_1 \cup E_2, O_1 \cup O_2) \enspace.
+\]
+Intuitively, this combination conveys the fact that there are no ordering constraints between the effects
+in the arguments graphs.
+
+Given two execution graphs $S_1 = (N_1, E_1, O_1)$ and $S_1 = (N_2, E_2, O_2)$ and an ordering label $l$,
+the ordered combination $\ordered{S_1}{l}{S_2}$ is defined as follows:
+\[
+  \ordered{S_1}{l}{S_2} \triangleq (N_1 \cup N_2, E_1 \cup E_2 \cup (O_1 \times N_2 \times \{l\}, O_2)) \enspace.
+\]
+Intuitively, this combination constrains the output effects of $S_1$ to appear before any effect of $S_2$ with respect
+to the given ordering label.
+
+\subsection{Input Configurations and Output Configurations}
+
+Input configurations modes are unique to the semantic relation that uses them.
```

Comment:
modes -> rule application domains?

what is an input configuration?

I think it would be worthwhile to recall the concepts in this sentence right here, and give a pointer to where they were first used